lib/proxy/util.rb

```diff
@@ -27,8 +28,13 @@ def start(&ensured_block)
           logger.info "[#{thr.pid}] Started task #{cmdline_string}"
           stdin.write(input) if input
           stdin.close
-          stdout.each do |line|
-            logger.debug "[#{thr.pid}] #{line}"
+          unless @output.nil?
+            File.binwrite(@output, stdout.read)
```

Comment:
Note this reads the whole file to memory and then writes it at once. It means that if there are large files to be extracted in the ISO the Smart Proxy's memory usage will spike. In the `Open3` module there are other methods which allow streaming.

Reply:
I guess this also what you refer to in your comment here?
> Perhaps ArchiveExtract shouldn't call super at all in start but rather implement its own logic.

Can you give a litte more detail what you have in mind with `own logic`? Should it be a class different from CommandTask? Or should there just be a different `initialize` when you want to pipe the output to a file? I guess we would also need a different (or at least more comprehensive) `start` method which differentiates between piping the output or just logging it.
Do you think about a more general implementation which pipes the output to another command or just into a file? I would favour the latter.

Thanks for your help on this, highly appreciated!